plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java

```diff
@@ -307,6 +311,52 @@ public void testMissingColumnsOnInsert()
         assertQuery("SELECT * FROM test_col_insert", "SELECT 1, 'val1', 'val2'");
     }
 
+    @Override
+    public void testTopNPushdown()
+    {
+        throw new SkipException("Phoenix does not support topN push down, but instead replaces partial topN with partial Limit.");
+    }
+
+    @Test
+    public void testReplacePartialTopNWithLimit()
+    {
+        assertThat(query("SELECT orderkey FROM orders ORDER BY orderkey LIMIT 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class, ExchangeNode.class, ExchangeNode.class, LimitNode.class);
+
+        assertThat(query("SELECT orderkey FROM orders ORDER BY orderkey DESC LIMIT 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class, ExchangeNode.class, ExchangeNode.class, LimitNode.class);
+
+        assertThat(query("SELECT * FROM orders ORDER BY orderkey, custkey LIMIT 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class, ExchangeNode.class, ExchangeNode.class, LimitNode.class);
+
+        assertThat(query("SELECT * FROM orders ORDER BY orderkey, custkey DESC LIMIT 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class, ExchangeNode.class, ExchangeNode.class, LimitNode.class);
+    }
+
+    /*
+     * Make sure that partial topN is replaced with a partial limit when the input is presorted.
+     */
+    @Test
+    public void testUseSortedPropertiesForPartialTopNElimination()
+    {
+        String tableName = "test_propagate_table_scan_sorting_properties";
+        // salting ensures multiple splits
+        String createTableSql = format("" +
+                        "CREATE TABLE %s WITH (salt_buckets = 5) AS " +
+                        "SELECT * FROM tpch.tiny.customer",
+                tableName);
+        assertUpdate(createTableSql, 1500L);
+
+        String expected = "SELECT custkey FROM customer ORDER BY 1 NULLS FIRST LIMIT 100";
```

Comment:
One doubt here, assuming that phoenix will honour whatever nulls sorting order Trino will pass down in the jdbc query to it, we can expect this to work even without the `NULLS FIRST` part right ?
The hive case is a bit different as there the sort order of `NULLS FIRST` is fixed when the sorted files are written.

Reply:
Yes. It will just be slower on the Phoenix side if there happens to an index on the sort column(s), The index can only be used with NULL FIRST. So for correctness this is not needed.